helper/Data.php

```diff
@@ -106,11 +106,19 @@ public function __construct(
     public function fetchPaymentMethods(Cart $cart, $language)
     {
         $merchantAccount = \Configuration::get('ADYEN_MERCHANT_ACCOUNT');
+        $customer = new \Customer($cart->id_customer);
 
         if (!$merchantAccount) {
             $this->logger->error(
                 "The merchant account field is empty, check your Adyen configuration in Prestashop."
             );
+
+            return array();
+        } elseif (!\Validate::isLoadedObject($customer)) {
```

Comment:
Instead of an else if just do these validations in order and since you already do an early return it's easier to read the code.

```suggestion
        }
        
        if (!\Validate::isLoadedObject($customer)) {
```